pkg/operator/ceph/csi/csi.go

```diff
@@ -31,7 +31,7 @@ import (
 func ValidateAndConfigureDrivers(context *clusterd.Context, namespace, rookImage, securityAccount string, serverVersion *version.Info, ownerRef *metav1.OwnerReference) {
 	var v *CephCSIVersion
 	var err error
-	if !AllowUnsupported {
+	if !AllowUnsupported && CSIEnabled() {
 		if v, err = validateCSIVersion(context.Clientset, namespace, rookImage, securityAccount, ownerRef); err != nil {
```

Comment:
Also if you could modify line 40 to something like:
```go
logger.Info("Skipping csi version check, since unsupported versions are allowed or csi is disabled")
```